cellprofiler/modules/colortogray.py

```diff
@@ -537,6 +536,6 @@ def upgrade_settings(self,
         for i in range(nchannels):
             idx = SLOT_FIXED_COUNT + SLOT_CHANNEL_CHOICE + i * SLOTS_PER_CHANNEL
             channel_idx = self.get_channel_idx_from_choice(setting_values[idx])
-            setting_values[idx] = self.channel_names[channel_idx]
+            setting_values[idx] = channel_idx+1
```

Comment:
I adapted the `upgrade_settings` by changing this line as well as the `get_channel_idx_from_choice` function.
This seemed to work fine for the example settings in the `test_colortogray` and some manual examples.

I noticed that the `upgrade_settings` for this module was a bit special, as it runs the `# Standardize the channel choices` ignoring the actual `variable_revision_number` - is there actually a particular reason for that? Or should I better also put this section in an `if` clause dependent on the  `variable_revision_number`?

Reply:
The purpose of the `upgrade_settings` is for allowing previous versions of `.cppipe` files to still be loaded into new version of cellprofiler. I'm not sure that this block is needed anymore at all, but I would test two things:
1) Export a `.cppipe` file with a bunch of channels in the master branch, then check to make sure that the pipeline loads properly in your modified branch
2) Perform the same steps as above, but remove that whole standardization block entirely! See what it changes :) It may not be needed at all

Thanks again for your work here!